src/compile/legend/parse.ts

```diff
@@ -209,7 +209,7 @@ function getProperty<K extends keyof LegendComponentProps>(
       ) as LegendComponentProps[K];
 
     case 'title':
-      return (fieldDefTitle(fieldDef, model.config, {allowDisabling: true}) || undefined) as LegendComponentProps[K];
+      return fieldDefTitle(fieldDef, model.config, {allowDisabling: true}) as LegendComponentProps[K];
```

Comment:
What? So does `fieldDefTitle` return null or ''? Should we refactor the function instead?

Reply:
Yes, but null or "" is fine.  Previously we only do this to minimize the Vega output so the title doesn't output anything, but it leads to confusion in the merging process.  At this point, it's not worth the simplification. That said, if you strongly think we should simplify the output, you can add some logic to assemble. (I'd rather not.)